### PR TITLE
add TLS support

### PR DESCRIPTION
**Pull request by fmoor**

### CODE REVIEW

**Comment by fmoor** [comment]
If we want to use `OptionalBool` for the `edgedb.Options.VerifyHostname` argument to `Connect()` then this PR is blocking on https://github.com/edgedb/edgedb-go/pull/136

**Comment by elprans** [comment]
Let's port the definition of optionals from #136 to unblock and avoid rushing #136.

**Review by fantix — APPROVED** [review]
I'm not familiar with Golang, but the logic looks correct to me. 👍

File: connect.go [hunk 1]
```diff
@@ -152,6 +153,14 @@ func (c *baseConn) connect(r *buff.Reader, cfg *connConfig) error {
 		}
 	}
 
+	_, isTLS := c.conn.(*tls.Conn)
+	if !isTLS && c.protocolVersion.gte(version{0, 11}) {
+		_ = c.close()
+		return &clientConnectionError{msg: fmt.Sprintf(
+			"server clamming to use protocol version %v.%v without using TLS",
```
Comment (elprans) [inline connect.go:160]:
```suggestion
			"server claims protocol version %v.%v without using TLS",
```

File: windows.go [hunk 2]
```diff
@@ -18,4 +18,12 @@
...
 package edgedb
 
+import "crypto/x509"
+
 var defaultHosts = []string{"localhost"}
+
+func getCertPool() (*x509.CertPool, error) {
+	// x509.SystemCertPool() doesn't work on Windows.
+	// https://github.com/golang/go/issues/16736
+	return x509.NewCertPool(), nil
```
Comment (elprans) [inline windows.go:28]: I think we should bundle the Mozilla CA bundle for parity with other bindings. certifi has a [copy](https://github.com/certifi/python-certifi/blob/master/certifi/cacert.pem)

File: connutils.go [hunk 3]
```diff
@@ -448,6 +490,54 @@ func parseConnectDSNAndArgs(
 		waitUntilAvailable = 30 * time.Second
 	}
 
+	roots, err := getCertPool()
```
Comment (fmoor) [inline connutils.go:493]: We should probably not include system certs when the user provides their own certificate right?

Comment (elprans) [reply connutils.go:493]: I think so.

Comment (fmoor) [reply connutils.go]: changed

File: windows.go (hunk 2 above)
Comment (fmoor) [reply windows.go]: changed

File: connect.go (hunk 1 above)
Comment (elprans) [reply connect.go:160]: This doesn't seem to have been changed in the latest diff.

**Review by elprans — APPROVED** [review]
LGTM!

**Pull request merged**